sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -150,9 +150,9 @@ void analyze(boolean includeRules, SensorContext context) {
     Path analysisOutput = protobufReportPathForMSBuild12(context);
 
     try {
-      Files.write(analysisInput, analysisSettings.getBytes(Charsets.UTF_8));
+      Files.write(analysisInput, analysisSettings.getBytes(StandardCharsets.UTF_8));
     } catch (IOException e) {
-      throw Throwables.propagate(e);
+      throw new RuntimeException(e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00112)